### PR TITLE
Adding @ts-check

### PR DESCRIPTION
**Pull request by isunitha98selvan**
Solved some of the errors which showed up due to adding "checkJS":true in jsconfig.json.
References #1905

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ export function calculateLeftPad(text) {
  *
  * @param {Object} opts Configure the resource hint.
  * @param {String} opts.hint The type of hint (see resourceHints).
- * @param {URL|String} opts.href The URL for the resource or origin.
+ * @param {String} opts.href The URL for the resource or origin.
```
Comment (saschanaz) [inline src/core/utils.js]: 👍 as no code really passes a `URL` object.

File: src/jsconfig.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "compilerOptions": {
     "target": "es2018",
     "moduleResolution": "node",
-    "noImplicitThis": true
+    "noImplicitThis": true,
+    "checkJs": true
```
Comment (saschanaz) [inline src/jsconfig.json]:
```suggestion
```

File: src/core/utils.js (hunk 1 above)
Comment (marcoscaceres) [reply src/core/utils.js]: Agree.

File: src/jsconfig.json (hunk 2 above)
Comment (isunitha98selvan) [reply src/jsconfig.json]: Resolved

File: src/w3c/seo.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ export function run(conf) {
   switch (conf.canonicalURI) {
     case "edDraft":
       if (conf.edDraftURI) {
-        conf.canonicalURI = new URL(conf.edDraftURI, document.location).href;
```
Comment (saschanaz) [inline src/w3c/seo.js]: Would you run `npm run lint -- --fix` and commit the result? Thanks!

Comment (isunitha98selvan) [reply src/w3c/seo.js]: Done

**Comment by saschanaz** [comment]
Thanks! 👍

**Pull request merged**